jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -284,6 +308,42 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
 
         return collapsed_spec
 
+    def cone_aperture(self):
+        if not self._app._jdaviz_helper._loaded_mask_cube:
+            snackbar_message = SnackbarMessage(
+                "Cannot create cone aperture without valid mask cube loaded.",
+                color="error",
+                sender=self)
+            self.hub.broadcast(snackbar_message)
+            return
+        # Retrieve flux cube and create an array to represent the cone mask
+        flux_cube = self._app._jdaviz_helper._loaded_flux_cube.get_object(cls=Spectrum1D,
+                                                                          statistic=None)
+        masks_boolean_values = np.zeros_like(flux_cube.flux.value, dtype=bool)
+
+        # Center is reverse coordinates
+        center = (self.aperture.selected_spatial_region.center.y,
+                  self.aperture.selected_spatial_region.center.x)
+        # TODO: Replace with code for retrieving display_unit in cubeviz when it is available
+        display_unit = flux_cube.spectral_axis.unit
+        if display_unit.physical_type != 'length':
+            error_msg = (f'Spectral axis unit physical type is '
+                         f'{display_unit.physical_type}, must be length')
+            snackbar_message = SnackbarMessage(error_msg, color="error", sender=self)
+            self.hub.broadcast(snackbar_message)
+            raise AttributeError(error_msg)
```

Comment:
I don't think there is a point to emit snackbar message if this is going to crash right after that. Either snackbar + no-op return *or* raise exception.